transports/p2p/p2psync/manager.go

```diff
@@ -606,7 +609,6 @@ func (sm *SyncManager) handleHeadersMsg(hmsg *headersMsg) {
 	// headers starting from the latest known header and ending with the
 	// next checkpoint.
 	sm.sendGetHeadersWithPassedParams([]*chainhash.Hash{finalHash}, sm.nextCheckpoint.Hash, peer)
```

Comment:
After last changes the app throws panic (after few seconds of working).

It is related to `finalHash`.

![image](https://github.com/bitcoin-sv/block-headers-service/assets/152964795/8eae4c6e-ef70-4618-b78a-f687036669c3)

![image](https://github.com/bitcoin-sv/block-headers-service/assets/152964795/a63ab11e-c028-4265-9ec3-b1ab638af333)

Reply:
fixed